pkg/store/consul/consultest/fake_preparer.go

```diff
@@ -78,6 +71,40 @@ func (f *FakePreparer) Enable() {
 					continue
 				}
 
+				realityPods := make(map[types.PodLocation]manifest.Manifest)
+				for _, pod := range allReality {
+					realityPods[pod.PodLocation] = pod.Manifest
+				}
+
+				// Set pods that are in intent
+				for podLocation, intentManifest := range intentPods {
+					intentSHA, err := intentManifest.SHA()
+					if err != nil {
+						f.logger.Errorf("error computing intent sha: %v", err)
+						continue
+					}
+
+					var realitySHA string
+					if realityManifest, ok := realityPods[podLocation]; ok {
+						realitySHA, err = realityManifest.SHA()
+						if err != nil {
+							f.logger.Errorf("error computing reality sha: %v", err)
+							continue
+						}
+					}
+
+					if realitySHA != intentSHA {
```

Comment:
Is this the only change that reduces the fake preparer's interaction with consul? I guess it reduces unnecessary idempotent writes.